core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -1102,110 +1102,6 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     }
   }
 
-  test("Completions in zombie tasksets update status of non-zombie taskset") {
```

Comment:
ok I'm playing with locally now and see this test fails ... I'm trying to figure out how to fix it now.

Reply:
I think adding this [here](https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala#L118), to the mocked DAGScheduler impl inside TaskSchedulerImplSuite, is what is needed to get this to pass (all tests within spark.scheduler pass for me with this change):

```scala
      override def taskEnded(
          task: Task[_],
          reason: TaskEndReason,
          result: Any,
          accumUpdates: Seq[AccumulatorV2[_, _]],
          taskInfo: TaskInfo): Unit = {
        if (reason == Success) {
          // For SPARK-23433 / SPARK-25250, need to make DAGScheduler lets all tasksets know
          // about complete partitions.  Super implementation is not enough, because we've mocked
          // out too much of the rest of the DAGScheduler.
          taskScheduler.completeTasks(task.partitionId, task.stageId, taskInfo)
        }
        super.taskEnded(task, reason, result, accumUpdates, taskInfo)
      }
```

Reply:
Thank you for providing the fix. Wasn't sure about keeping this test before hence, I removed it, but have added the test back.